pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -492,13 +493,19 @@ func (c *clustersReconciler) runGC(ctx context.Context) {
 			return
 		case <-time.After(c.gcInterval):
 			log.V(4).Info("Run Garbage Collection for Lost Remote Workloads")
-			for clusterName, rc := range c.remoteClients {
-				rc.runGC(ctrl.LoggerInto(ctx, log.WithValues("multiKueueCluster", clusterName)))
+			for _, rc := range c.getRemoteClients() {
+				rc.runGC(ctrl.LoggerInto(ctx, log.WithValues("multiKueueCluster", rc.clusterName)))
 			}
 		}
 	}
 }
 
+func (c *clustersReconciler) getRemoteClients() []*remoteClient {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	return maps.Values(c.remoteClients)
```

Comment:
Let's instead add a method in pkg/util/maps.

The `exp` package doesn't have backward compatibility guarantees.

Reply:
Thanks. Added.